### PR TITLE
Rc app organization

### PR DESCRIPTION
**Pull request by ryancoughlin**

### CODE REVIEW

File: app/components/app.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export default class App extends Component {
+  componentDidMount() {
+    SplashScreen.hide()
+  }
+
+  componentWillUnmount() {
+    AppState.removeEventListener('change', this.handleAppStateChange)
+  }
+
+  handleAppStateChange = (appState) => {
```
Comment (BlakeWilliams) [inline app/components/app.js]: does this accept `prevState`? That might be better than using `this.state` to store it

File: app/components/station-detail/index.js [hunk 2]
```diff
@@ -7,12 +7,22 @@ import * as actions from '../../actions/station'
...
 import SaveLocationButton from '../buttons/save-location-button'
 import RemoveLocationButton from '../buttons/remove-location-button'
 import NavigationBarItem from '../buttons/navigation-bar-item'
 
 const StationDetail = class extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      tideListVisibility: false,
```
Comment (BlakeWilliams) [inline app/components/station-detail/index.js]: `tideListVisible` so it reads more like a command than a state of being?

Comment (ryancoughlin) [inline app/components/station-detail/index.js]: Yup! this changed in a later commit.

File: app/components/app.js (hunk 1 above)
Comment (ryancoughlin) [inline app/components/app.js]: What do you mean here?

Comment (BlakeWilliams) [inline app/components/app.js]: nevermind, I thought the api passed previous state for some reason

**Pull request merged**